tests/testsuite/cargo_add/no_public/in/Cargo.toml

```diff
@@ -1,6 +1,9 @@
 cargo-features = ["public-dependency"]
 [workspace]
 
+[lib]
+path = "../../add_basic.in/src/lib.rs"
```

Comment:
By specifing `path = "../../add_basic.in/src/lib.rs"`,  i suppose this could reduce a lot of same file.  
  
the inspiration was from `no_optional` and `optional`, they made a symbolic link to the `add-basic.in`
```
ll optional/in no_optional/in
lrwxrwxrwx 1 root root 15 Oct  9 15:52 no_optional/in -> ../add-basic.in/
lrwxrwxrwx 1 root root 15 Oct  9 15:52 optional/in -> ../add-basic.in/
```

If there is a need to keep each test case separate, I can create a separate lib.rs for each test case in next commit.